jetty-core/jetty-util/src/main/java/org/eclipse/jetty/util/resource/PathResource.java

```diff
@@ -371,6 +352,25 @@ private static URI toUri(Path path)
         return pathUri;
     }
 
+    @Override
+    public boolean equals(Object obj)
+    {
+        if (this == obj)
+            return true;
+        if (obj == null)
+            return false;
+        if (getClass() != obj.getClass())
+            return false;
+        PathResource other = (PathResource)obj;
+        return Objects.equals(path, other.path) && Objects.equals(uri, other.uri);
```

Comment:
Why do we need equals/hashCode at all?

The comparison on `uri` is odd, that's a user-space provided field.

Reply:
I'm not sure why we need to compare `PathResource`, I know there are tests for the `equals()` but not sure if we actually use it anywhere. But to compare these correctly you need to take the `uri` into account.

You can come up with examples where two different `PathResource`s have the same `path` field but one is classified as an alias and the other is not. So comparing only on `path` isn't quite correct.

Reply:
This will impact the ResourceService cache though.

The same Resource will exist on multiple locations once this new equals/hashcode exists.
Is that an acceptable behavior?